src/zenml/models/v2/base/scoped.py

```diff
@@ -347,3 +519,79 @@ def get_custom_filters(self) -> List["ColumnElement[bool]"]:
             )
 
         return custom_filters
+
+    def apply_sorting(
+        self,
+        query: AnyQuery,
+        table: Type["AnySchema"],
+    ) -> AnyQuery:
+        """Apply sorting to the query.
+
+        Args:
+            query: The query to which to apply the sorting.
+            table: The query table.
+
+        Returns:
+            The query with sorting applied.
+        """
+        from sqlmodel import asc, desc, func
+
+        from zenml.enums import SorterOps, TaggableResourceTypes
+        from zenml.zen_stores.schemas import (
+            ArtifactSchema,
+            ArtifactVersionSchema,
+            ModelSchema,
+            ModelVersionSchema,
+            PipelineRunSchema,
+            PipelineSchema,
+            RunTemplateSchema,
+            TagResourceSchema,
+            TagSchema,
+        )
+
+        sort_by, operand = self.sorting_params
+
+        if sort_by == "tag":
+            resource_type_mapping = {
+                ArtifactSchema: TaggableResourceTypes.ARTIFACT,
+                ArtifactVersionSchema: TaggableResourceTypes.ARTIFACT_VERSION,
+                ModelSchema: TaggableResourceTypes.MODEL,
+                ModelVersionSchema: TaggableResourceTypes.MODEL_VERSION,
+                PipelineSchema: TaggableResourceTypes.PIPELINE,
+                PipelineRunSchema: TaggableResourceTypes.PIPELINE_RUN,
+                RunTemplateSchema: TaggableResourceTypes.RUN_TEMPLATE,
+            }
+
+            query = (
+                query.outerjoin(
+                    TagResourceSchema,
+                    (table.id == TagResourceSchema.resource_id)
+                    & (
+                        TagResourceSchema.resource_type
+                        == resource_type_mapping[table]
+                    ),
+                )
+                .outerjoin(TagSchema, TagResourceSchema.tag_id == TagSchema.id)
+                .group_by(table.id)
+            )
+
+            if operand == SorterOps.ASCENDING:
+                query = query.order_by(
+                    asc(
+                        func.group_concat(TagSchema.name, ",").label(
+                            "tags_list"
```

Comment:
Exactly. I realize that it might not be the prettiest solution. That's why I also had a brief discussion with @znegrin about it and we decided to implement it like this.